semgrep.yml

```diff
@@ -147,6 +147,18 @@ rules:
     paths:
       include:
         - cli/src/*
+  - id: reraise-timeout
+    patterns:
+      - pattern: try ... with | $X -> ...
```

Comment:
I think we want to mainly detect the 'try with' where we "swallow" an exn with a general pattern
like
try ... with | exn -> ...
That's where it's dangerous because the Timeout would not pop up. With 'try ... with Not_found -> ... it's fine. It will still pop up.

Reply:
But maybe we can be ugly and check that the $X is an ident with metavariable-regex:?
Still I'm not sure it would catch code like try foo with | Not_found -> bla | OtherExn -> bla | exn -> dangerous.

Reply:
Oh right, good point. I'll see if I can come up with a pattern that will match what we want.